### PR TITLE
feat: add SIG team labels to review labels

### PR DESCRIPTION
**Pull request by myakove**
Enhanced review label handling to include SIG team information:

- Added teams_and_members property to OwnersFileHandler to map OWNERS file directories to their team members (approvers + reviewers combined)
- Added get_user_sig_suffix() method to generate SIG label suffix based on user's team membership
- Modified LabelsHandler to append SIG suffix to review labels (e.g., approved-john becomes approved-john-sig-storage)
- Added comprehensive tests for teams_and_members property and get_user_sig_suffix() method

This provides better visibility into which SIG teams approved/reviewed PRs.

## Summary by CodeRabbit

* **New Features**
  * Reviewer labels can include per-user SIG suffixes when a repository SIG file is configured; repository config adds a sig-labels option.

* **Documentation**
  * Added SIG Labels for Review Tracking docs and updated example config to show sig-labels usage.

* **Chores**
  * Standardized logging format and improved error handling and file-read fallbacks.

* **Tests**
  * Expanded tests for SIG file loading, per-user suffix generation, and label behavior across many edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: webhook_server/libs/handlers/owners_files_handler.py [hunk 1]
```diff
@@ -75,6 +76,64 @@ def root_approvers(self) -> list[str]:
...
+        for team_name, members in self.teams_and_members.items():
+            if username in members:
+                sig_names.append(team_name)
+
+        sig_names.sort()
+
+        if sig_names:
+            return "-" + "-".join(sig_names)
+        return ""
+
```
[the remark this answers is not in the document]
Comment (myakove) [reply webhook_server/libs/handlers/owners_files_handler.py]: Done

File: webhook_server/libs/handlers/owners_files_handler.py [hunk 2]
```diff
@@ -117,18 +176,18 @@ async def list_changed_files(self, pull_request: PullRequest) -> list[str]:
...
-                self.logger.error(f"{self.log_prefix} git diff command failed")
+                self.logger.error("%s git diff command failed", self.log_prefix)
                 return []
 
             # Parse output: split by newlines and filter empty lines
             changed_files = [line.strip() for line in out.splitlines() if line.strip()]
 
-            self.logger.debug(f"{self.log_prefix} Changed files: {changed_files}")
+            self.logger.debug("%s Changed files: %s", self.log_prefix, changed_files)
             return changed_files
```
[the remark this answers is not in the document]
Comment (myakove) [reply webhook_server/libs/handlers/owners_files_handler.py]: Done
